src/eeprom.c

```diff
@@ -0,0 +1,82 @@
+#include "eeprom.h"
+
+#define BYTES_PER_WORD 4
+
+/**
+ * @brief Function to read eeprom data from a specific address
+ *
+ * @param[in] start_address Address to read from
+ * @param[in] num_elements Number of bytes to be read
+ * @param[out] output_data Pointer to a word to save the read data
+ */
+void eeprom_read_data(uint32_t start_address, uint16_t num_elements,
+		      uint8_t *output_data)
+{
+	uint16_t iter;
+	uint32_t *memory_ptr = (uint32_t *)start_address;
+
+	for (iter = 0; iter < num_elements / BYTES_PER_WORD; iter++) {
+		*(uint32_t *)output_data = *(memory_ptr + iter);
+		output_data += BYTES_PER_WORD;
+	}
+}
+
+/**
+ * @brief Function to save data on a page of EEPROM
+ *
+ * - Unlock flash
+ * - Erase page
+ * - Program flash memory word by word (32-bits )and verify that it is written
+ *
+ * @param[in] page_address Address of a EEPROM page to flash on
+ * @param[in] input_data Pointer to the data to be flashed
+ * @param[in] num_elements Number of bytes to be flashed
+ * @return Flash state.
+ */
+uint32_t eeprom_flash_page(uint32_t page_address, uint8_t *input_data,
+			   uint16_t num_elements)
+{
+	uint16_t iter;
+	uint32_t flash_status = 0;
+
+	flash_unlock();
+
+	flash_erase_page(page_address);
+	flash_status = flash_get_status_flags();
+	if (flash_status != FLASH_SR_EOP)
+		return flash_status;
+
+	for (iter = 0; iter < num_elements; iter += BYTES_PER_WORD) {
+		flash_program_word(page_address + iter,
+				   *((uint32_t *)(input_data + iter)));
+		flash_status = flash_get_status_flags();
+		if (flash_status != FLASH_SR_EOP)
+			return flash_status;
+
+		if (*((uint32_t *)(page_address + iter)) !=
+		    *((uint32_t *)(input_data + iter)))
+			return FLASH_WRONG_DATA_WRITTEN;
+	}
+
+	return RESULT_OK;
+}
```

Comment:
I do not like the `BYTES_PER_WORD` use here very much...

Reply:
You are right, in fact I have created issue #328 to rethink EEPROM inputs